class-sc-wp-updates-notifier.php

```diff
@@ -421,24 +433,65 @@ public function check_themes_against_notified( $themes_need_update ) {
 
 
 		/**
-		 * Sends email notification.
+		 * Sends email message.
 		 *
 		 * @param string $message Holds message to be sent in body of email.
 		 *
-		 * @return void
+		 * @return bool Whether the email contents were sent successfully.
 		 */
-		public function send_notification_email( $message ) {
+		public function send_email_message( $message ) {
 			$settings = $this->get_set_options( self::OPT_FIELD ); // get settings
-			$subject  = sprintf( __( 'WP Updates Notifier: Updates Available @ %s', 'wp-updates-notifier' ), home_url() );
+	
+			$subject = sprintf( __( 'WP Updates Notifier: Updates Available @ %s', 'wp-updates-notifier' ), home_url() );
 			add_filter( 'wp_mail_from', array( $this, 'sc_wpun_wp_mail_from' ) ); // add from filter
 			add_filter( 'wp_mail_from_name', array( $this, 'sc_wpun_wp_mail_from_name' ) ); // add from name filter
 			add_filter( 'wp_mail_content_type', array( $this, 'sc_wpun_wp_mail_content_type' ) ); // add content type filter
 			// phpcs:disable WordPressVIPMinimum.Functions.RestrictedFunctions.wp_mail_wp_mail
-			wp_mail( $settings['notify_to'], apply_filters( 'sc_wpun_email_subject', $subject ), apply_filters( 'sc_wpun_email_content', $message ) ); // send email
+			$response = wp_mail( $settings['notify_to'], apply_filters( 'sc_wpun_email_subject', $subject ), apply_filters( 'sc_wpun_email_content', $message ) ); // send email
 			// phpcs:enable
 			remove_filter( 'wp_mail_from', array( $this, 'sc_wpun_wp_mail_from' ) ); // remove from filter
 			remove_filter( 'wp_mail_from_name', array( $this, 'sc_wpun_wp_mail_from_name' ) ); // remove from name filter
 			remove_filter( 'wp_mail_content_type', array( $this, 'sc_wpun_wp_mail_content_type' ) ); // remove content type filter
+
+			return $response;
+		}
+
+
+		/**
+		 * Sends slack post.
+		 *
+		 * @param string $message Holds message to be posted to slack.
+		 *
+		 * @return bool Success or failure.
+		 */
+		public function send_slack_message( $message ) {
+			$settings = $this->get_set_options( self::OPT_FIELD ); // get settings
+
+			$payload = array(
+				'username'   => 'WP Updates Notifier',
+				'icon_emoji' => ':robot_face:',
+				'text'       => $message,
+			);
+
+			if ( ! empty( $settings['slack_channel_override'] ) && '' !== $settings['slack_channel_override'] ) {
+				$payload['channel'] = $settings['slack_channel_override'];
+			}
+
+			$response = wp_remote_post(
+				$settings['slack_webhook_url'], 
+				array(
+					'method' => 'POST',
+					'body'   => array(
+						'payload' => wp_json_encode( $payload ),
+					),
+				) 
+			);
+
+			if ( is_wp_error( $response ) ) {
+				return false;
+			} else {
+				return true;
+			}
```

Comment:
If this is the only return value you could clean this up as `return is_wp_error( $response )` I would think.